### PR TITLE
Change vote and summary percentages to xx.dd% format

### LINKED ISSUE #159: Change vote percentage and summary percentages to "xx.dd%" format
$title.

### PR DESCRIPTION
**Pull request by MaryamZi**
$title.

This PR also removes retry from the client temporarily. 

Fixes #159

### CODE REVIEW

File: distributor/src/distributor/genhtml.bal [hunk 1]
```diff
@@ -57,14 +57,14 @@ function generatePresidentialResultHtml(string electionCode, map<json> result, b
     foreach json j in partyResults {
         map<json> pr = <map<json>> j; // value is a json object
         if firstRound {
-            body += "<tr><td>" + <string>pr.candidate + "</td><td class='text-center'>" + <string>pr.party_code + "</td><td class='text-right'>" + commaFormatInt(<int>pr.vote_count) + "</td><td class='text-right'>" + <string>pr.vote_percentage + "%</td></tr>";
+            body += "<tr><td>" + <string>pr.candidate + "</td><td class='text-center'>" + <string>pr.party_code + "</td><td class='text-right'>" + commaFormatInt(<int>pr.vote_count) + "</td><td class='text-right'>" + <string>pr.vote_percentage + "</td></tr>";
```
Comment (chamil321) [inline distributor/src/distributor/genhtml.bal]: Shall we not change this presidential code?

File: distributor/src/distributor/genhtml.bal [hunk 2]
```diff
@@ -57,14 +57,14 @@ function generatePresidentialResultHtml(string electionCode, map<json> result, b
...
         } else {
             body += "<tr><td>" + <string>pr.candidate + "</td><td class='text-center'>" + <string>pr.party_code 
                                + "</td><td class='text-right'>" + commaFormatInt(<int>pr.votes1st) 
                                + "</td><td class='text-right'>" + commaFormatInt(<int>pr.votes2nd) 
                                + "</td><td class='text-right'>" + commaFormatInt(<int>pr.votes3rd) 
                                + "</td><td class='text-right'>" + commaFormatInt(<int>pr.vote_count) 
-                               + "</td><td class='text-right'>" + <string>pr.vote_percentage + "%</td></tr>";
+                               + "</td><td class='text-right'>" + <string>pr.vote_percentage + "</td></tr>";
```
Comment (chamil321) [inline distributor/src/distributor/genhtml.bal]: This one too. Related to presidential E

File: distributor/src/distributor/genhtml.bal [hunk 3]
```diff
@@ -73,13 +73,13 @@ function generatePresidentialResultHtml(string electionCode, map<json> result, b
...
-            "    </div><div class='col-md-2 text-right'>" + <string>result.summary.percent_valid + "%</div><div class='col-md-2'></div>";
+            "    </div><div class='col-md-2 text-right'>" + <string>result.summary.percent_valid + "</div><div class='col-md-2'></div>";
     body += "  <div class='col-md-4 col-md-offset-2'>Rejected Votes</div>" + 
             "    <div class='col-md-2 text-right'>" + commaFormatInt(<int>result.summary.rejected) + 
-            "    </div><div class='col-md-2 text-right'>" + <string>result.summary.percent_rejected + "%</div><div class='col-md-2'></div>";
+            "    </div><div class='col-md-2 text-right'>" + <string>result.summary.percent_rejected + "</div><div class='col-md-2'></div>";
     body += "  <div class='col-md-4 col-md-offset-2'>Total Polled</div>" + 
             "    <div class='col-md-2 text-right'>" + commaFormatInt(<int>result.summary.polled) + 
-            "    </div><div class='col-md-2 text-right'>" + <string>result.summary.percent_polled + "%</div><div class='col-md-2'></div>";
+            "    </div><div class='col-md-2 text-right'>" + <string>result.summary.percent_polled + "</div><div class='col-md-2'></div>";
```
Comment (chamil321) [inline distributor/src/distributor/genhtml.bal]: These too

File: distributor/src/distributor/save.bal [hunk 4]
```diff
@@ -300,12 +300,10 @@ function addToPresidentialCumulative(map<json> jm) returns PresidentialCumulativ
...
-        // don't add up electors from postal PDs as those are already in the district elsewhere
-        string pdCode = <string>jm.pd_code; // check 
         accum.summary.electors += <int>jm.summary.electors;
-        accum.summary.percent_valid = (accum.summary.polled == 0) ? "0.00" : io:sprintf("%.2f", accum.summary.valid*100.0/accum.summary.polled);
-        accum.summary.percent_rejected = (accum.summary.polled == 0) ? "0.00" : io:sprintf("%.2f", accum.summary.rejected*100.0/accum.summary.polled);
-        accum.summary.percent_polled = (accum.summary.electors == 0) ? "0.00" : io:sprintf("%.2f", accum.summary.polled*100.0/accum.summary.electors);
+        accum.summary.percent_valid = (accum.summary.polled == 0) ? "0.00%" : string `${io:sprintf("%.2f", accum.summary.valid*100.0/accum.summary.polled)}%`;
+        accum.summary.percent_rejected = (accum.summary.polled == 0) ? "0.00%" : string `${io:sprintf("%.2f", accum.summary.rejected*100.0/accum.summary.polled)}%`;
+        accum.summary.percent_polled = (accum.summary.electors == 0) ? "0.00%" : string `${io:sprintf("%.2f", accum.summary.polled*100.0/accum.summary.electors)}%`;
```
Comment (chamil321) [inline distributor/src/distributor/save.bal]: These too

File: distributor/src/distributor/save.bal [hunk 5]
```diff
@@ -323,7 +321,7 @@ function addToPresidentialCumulative(map<json> jm) returns PresidentialCumulativ
                 accum.by_party[i]["votes2nd"] = (accum.by_party[i]["votes2nd"] ?: 0) + <int>pr[i].votes2nd;
                 accum.by_party[i]["votes3rd"] = (accum.by_party[i]["votes3rd"] ?: 0) + <int>pr[i].votes3rd;
             }
-            accum.by_party[i].vote_percentage = (accum.summary.valid == 0) ? "0.00" : io:sprintf ("%.2f", ((accum.by_party[i].vote_count*100.0)/accum.summary.valid));
+            accum.by_party[i].vote_percentage = (accum.summary.valid == 0) ? "0.00%" : string `${io:sprintf ("%.2f", ((accum.by_party[i].vote_count*100.0)/accum.summary.valid))}%`;
```
Comment (chamil321) [inline distributor/src/distributor/save.bal]: this too

File: subscriber/src/subscriber/genhtml.bal [hunk 6]
```diff
@@ -48,21 +48,21 @@ function generatePresidentialResultHtml(string electionCode, map<json> result, b
...
-            "    </div><div class='col-md-2 text-right'>" + <string>result.summary.percent_valid + "%</div><div class='col-md-2'></div>";
+            "    </div><div class='col-md-2 text-right'>" + <string>result.summary.percent_valid + "</div><div class='col-md-2'></div>";
     body += "  <div class='col-md-4 col-md-offset-2'>Rejected Votes</div>" + 
             "    <div class='col-md-2 text-right'>" + commaFormatInt(<int>result.summary.rejected) + 
-            "    </div><div class='col-md-2 text-right'>" + <string>result.summary.percent_rejected + "%</div><div class='col-md-2'></div>";
+            "    </div><div class='col-md-2 text-right'>" + <string>result.summary.percent_rejected + "</div><div class='col-md-2'></div>";
     body += "  <div class='col-md-4 col-md-offset-2'>Total Polled</div>" + 
             "    <div class='col-md-2 text-right'>" + commaFormatInt(<int>result.summary.polled) + 
-            "    </div><div class='col-md-2 text-right'>" + <string>result.summary.percent_polled + "%</div><div class='col-md-2'></div>";
+            "    </div><div class='col-md-2 text-right'>" + <string>result.summary.percent_polled + "</div><div class='col-md-2'></div>";
```
Comment (chamil321) [inline subscriber/src/subscriber/genhtml.bal]: These too

File: testdriver/src/testdriver/sendresults.bal [hunk 7]
```diff
@@ -279,13 +279,13 @@ function createNationalResult (string resultType, map<map<json>>[] results, map<
...
+        natByParty[i]["vote_percentage"] = (natSummary.valid == 0) ? "0.00%" : string `${io:sprintf("%.2f", votes_by_party[i]*100.0/natSummary.valid)}%`;
     }
 
     // set the percentages in the summary
-    natSummary.percent_valid = (natSummary.polled == 0) ? "0.00" : io:sprintf("%.2f", natSummary.valid*100.0/natSummary.polled);
-    natSummary.percent_rejected = (natSummary.polled == 0) ? "0.00" : io:sprintf("%.2f", natSummary.rejected*100.0/natSummary.polled);
-    natSummary.percent_polled = (natSummary.electors == 0) ? "0.00" : io:sprintf("%.2f", natSummary.polled*100.0/natSummary.electors);
+    natSummary.percent_valid = (natSummary.polled == 0) ? "0.00%" : string `${io:sprintf("%.2f", natSummary.valid*100.0/natSummary.polled)}%`;
+    natSummary.percent_rejected = (natSummary.polled == 0) ? "0.00%" : string `${io:sprintf("%.2f", natSummary.rejected*100.0/natSummary.polled)}%`;
+    natSummary.percent_polled = (natSummary.electors == 0) ? "0.00%" : string `${io:sprintf("%.2f", natSummary.polled*100.0/natSummary.electors)}%`;
```
Comment (chamil321) [inline testdriver/src/testdriver/sendresults.bal]: Shall we not change this file also as it related to presidential E

**Comment by MaryamZi** [comment]
@chamil321, addressed all the suggestions, thanks!

File: distributor/src/distributor/save.bal [hunk 8]
```diff
@@ -300,8 +300,6 @@ function addToPresidentialCumulative(map<json> jm) returns PresidentialCumulativ
         accum.summary.valid += <int>jm.summary.valid;
         accum.summary.rejected += <int>jm.summary.rejected;
         accum.summary.polled += <int>jm.summary.polled;
-        // don't add up electors from postal PDs as those are already in the district elsewhere
-        string pdCode = <string>jm.pd_code; // check
```
Comment (chamil321) [inline distributor/src/distributor/save.bal:304]: Do we need to remove this?

Comment (MaryamZi) [reply distributor/src/distributor/save.bal:304]: Yeah it's not used anymore. The logic that used this variable has been removed, but seems like the variable wasn't.

**Review by chamil321 — APPROVED** [review]

**Pull request merged**